### PR TITLE
Refactor: Security policy

### PR DESCRIPTION
**Pull request by grignards**
See #6233

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- ~~[ ] Manual, quality tests have been done~~ => no manual test needed as this is a refactor and the automated test coverage is enough to catch errors.

### CODE REVIEW

**Review by Millefeuille42 — APPROVED** [review]

**Review by floreal15 — APPROVED** [review]

File: saml.ini
Comment (Millefeuille42) [inline saml.ini:1]: This file seems out of place, or is it ? I don't see any usage of it yet

**Comment by Millefeuille42** [comment]
Will this close #6207 ?

**Comment by grignards** [comment]
> Will this close #6207 ?

I think #6207 can be closed, the results of the investigation are in my comments that you reviewed. This PR is not about sharing behavior with LDAP/remote auth, only ensuring that using security policies to implement the SAML authentication is possible.

**Comment by Millefeuille42** [comment]
My bad... I mean #6208 but I guess both can be marked as done

**Comment by grignards** [comment]
> My bad... I meant #6208 but I guess both can be marked as done

For 6208, yes this PR is addressing it (as mentioned in the description).

**Pull request merged**